server/executor/runner.go

```diff
@@ -135,10 +134,15 @@ func (r persistentRunner) processExecQueue(job execReq) {
 
 	job.test = newTest
 
-	response, err := triggerer.Trigger(job.ctx, job.test)
+	traceID := model.IDGen.TraceID()
+	run.TraceID = traceID
+	r.handleDBError(r.updater.Update(job.ctx, run))
+
+	response, err := triggerer.Trigger(job.ctx, job.test, &trigger.TriggerOptions{
+		TraceID: traceID,
+	})
```

Comment:
I had to change that because the traceID would only be saved **after** the initial span was sent. This could cause race conditions (and I experienced a problem with it). So, by saving the run first and then triggering the transaction (injecting the same traceID saved into the run) we ensure that at the moment spans get sent to the otlp endpoint, we know the traceID and can add the spans to the right test run.